hummingbot/connector/exchange/ftx/ftx_exchange.pyx

```diff
@@ -0,0 +1,898 @@
+                "side": "buy" if is_buy else "sell",
+                "price": None,
+                "type": "market",
+                "size": amount,
+                "reduceOnly": False,
+                "ioc": False,
+                "postOnly": False,
+                "clientId": str(order_id),
+            }
+        else:
+            raise ValueError(f"Unknown order_type for FTX: {order_type}")
+
+        api_response = await self._api_request("POST", path_url=path_url, body=body)
+
+        return api_response
+
+    async def execute_buy(self,
+                          order_id: str,
+                          trading_pair: str,
+                          amount: Decimal,
+                          order_type: OrderType,
+                          price: Optional[Decimal] = s_decimal_0):
+        cdef:
+            TradingRule trading_rule = self._trading_rules[trading_pair]
+            double quote_amount
+            object decimal_amount
+            object decimal_price
+            str exchange_order_id
+            object tracked_order
+
+        decimal_amount = self.c_quantize_order_amount(trading_pair, amount)
+        decimal_price = (self.c_quantize_order_price(trading_pair, price)
+                         if order_type.is_limit_type()
+                         else s_decimal_0)
+
+        if decimal_amount < trading_rule.min_order_size:
+            raise ValueError(f"Buy order amount {decimal_amount} is lower than the minimum order size "
+                             f"{trading_rule.min_order_size}.")
+
+        try:
+            order_result = None
+            self.c_start_tracking_order(
+                order_id,
+                None,
+                trading_pair,
+                order_type,
+                TradeType.BUY,
+                decimal_price,
+                decimal_amount
+            )
+            if order_type.is_limit_type():
+                order_result = await self.place_order(order_id,
+                                                      trading_pair,
+                                                      decimal_amount,
+                                                      True,
+                                                      order_type,
+                                                      decimal_price)
+            elif order_type is OrderType.MARKET:
+                order_result = await self.place_order(order_id,
+                                                      trading_pair,
+                                                      decimal_amount,
+                                                      True,
+                                                      order_type,
+                                                      None)
+
+            else:
+                raise ValueError(f"Invalid OrderType {order_type}. Aborting.")
+
+            exchange_order_id = str(order_result["result"]["id"])
```

Comment:
Here an if-clause is needed to capture order_result["success"]==False to avoid error mentioned by @rxlxrxsx with message:
hummingbot.connector.exchange.ftx.ftx_exchange.FtxExchange.execute_sell
    exchange_order_id = str(order_result["result"]["id"])
KeyError: 'result'